### PR TITLE
AST-1043 - Add To Cart - Button Size option

### PR DESCRIPTION
**Pull request by GauravKhupse**
### Description
- Responsive Button Width option for Single Product.

### Screenshots
- https://share.bsf.io/7KuAQKBq

### Types of changes

New feature (non-breaking change which adds functionality

### How has this been tested?
Added in the card.

### Checklist:
- [x] My code is tested
- [x] My code passes the PHPCS tests
- [ ] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Comment by premanshup** [comment]
@GauravKhupse Code analysis is failing, please check.

File: inc/lib/astra-notices/class-astra-notices.php [hunk 1]
```diff
@@ -116,7 +116,7 @@ public function dismiss_notice() {
 			}
 
 			if ( false === wp_verify_nonce( $nonce, 'astra-notices' ) ) {
-				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.', 'astra' ) );
+				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.' ) );
```
Comment (premanshup) [inline inc/lib/astra-notices/class-astra-notices.php:119]:
```suggestion
				wp_send_json_error( esc_html_e( 'WordPress Nonce not validated.', 'astra' ) );
```

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 2]
```diff
@@ -1665,7 +1710,7 @@ public function add_styles() {
 				),
 			);
 			/* Parse CSS from array()*/
-			$yith_wcwl_main_style .= astra_parse_css( $yith_wcwl_main_style_small, '', astra_get_tablet_breakpoint() );
+			$yith_wcwl_main_style .= astra_parse_css( $yith_wcwl_main_style_small, astra_get_tablet_breakpoint() );
```
Comment (premanshup) [inline inc/compatibility/woocommerce/class-astra-woocommerce.php:1713]:
```suggestion
			$yith_wcwl_main_style .= astra_parse_css( $yith_wcwl_main_style_small, '', astra_get_tablet_breakpoint() );
```
@GauravKhupse Any reason for this change?

**Review by premanshup — CHANGES_REQUESTED** [review]

File: inc/lib/astra-notices/class-astra-notices.php (hunk 1 above)
Comment (GauravKhupse) [reply inc/lib/astra-notices/class-astra-notices.php]: Changes done!

File: assets/js/unminified/customizer-preview.js [hunk 3]
```diff
@@ -1841,6 +1841,32 @@ function hasWordPressWidgetBlockEditor() {
 				}
 			} );
 		} );
+		
+		wp.customize('astra-settings[single-product-cart-button-width]', function (value) {
+			value.bind(function (size) {
+				console.log(size);
```
[the remark this answers is not in the document]
Comment (GauravKhupse) [reply assets/js/unminified/customizer-preview.js]: Changes done!

File: inc/compatibility/woocommerce/class-astra-woocommerce.php [hunk 4]
```diff
@@ -1309,6 +1314,45 @@ public function add_styles() {
...
+			 */
+
+			$single_product_cart_button_width = astra_get_option( 'single-product-cart-button-width' );
+
+			$single_product_cart_button_width_desktop = ( isset( $single_product_cart_button_width ) && isset( $single_product_cart_button_width['desktop'] ) && ! empty( $single_product_cart_button_width['desktop'] ) ) ? $single_product_cart_button_width['desktop'] : '';
+
+			$single_product_cart_button_width_tablet = ( isset( $single_product_cart_button_width ) && isset( $single_product_cart_button_width['tablet'] ) && ! empty( $single_product_cart_button_width['tablet'] ) ) ? $single_product_cart_button_width['tablet'] : '';
+
+			$single_product_cart_button_width_mobile = ( isset( $single_product_cart_button_width ) && isset( $single_product_cart_button_width['mobile'] ) && ! empty( $single_product_cart_button_width['mobile'] ) ) ? $single_product_cart_button_width['mobile'] : '';
+
```
[the remark this answers is not in the document]
Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: Changes done!

File: inc/compatibility/woocommerce/class-astra-woocommerce.php (hunk 2 above)
Comment (GauravKhupse) [reply inc/compatibility/woocommerce/class-astra-woocommerce.php]: This is showing code analysis changes but now I have fixed using ignore the rule.

**Review by premanshup — APPROVED** [review]

**Review by imnavanath — APPROVED** [review]

**Pull request merged**